In [86]:
# Importing required libraries
import pandas as pd
import numpy as np
import datetime as dt

In [2]:
# Read the dataset using read_csv function of pandas
df = pd.read_csv("C:\\Users\\daivi\\Desktop\\CBD 2214 Big Data Fundamentals\\In Class Assignment\\disney_plus_titles.csv")

In [3]:
# To view the top 5 rows of our dataset
df.head()

show_id     type                                             title  \
0      s1    Movie  Duck the Halls: A Mickey Mouse Christmas Special   
1      s2    Movie                            Ernest Saves Christmas   
2      s3    Movie                      Ice Age: A Mammoth Christmas   
3      s4    Movie                        The Queen Family Singalong   
4      s5  TV Show                             The Beatles: Get Back   

                            director  \
0  Alonso Ramirez Ramos, Dave Wasson   
1                        John Cherry   
2                       Karen Disher   
3                    Hamish Hamilton   
4                                NaN   

                                                cast        country  \
0  Chris Diamantopoulos, Tony Anselmo, Tress MacN...            NaN   
1           Jim Varney, Noelle Parker, Douglas Seale            NaN   
2  Raymond Albert Romano, John Leguizamo, Denis L...  United States   
3  Darren Criss, Adam Lambert, Derek Hough, Alexa...            NaN   
4  John Lennon, Paul McCartney, George Harrison, ...            NaN   

          date_added  release_year rating  duration  \
0  November 26, 2021          2016   TV-G    23 min   
1  November 26, 2021          1988     PG    91 min   
2  November 26, 2021          2011   TV-G    23 min   
3  November 26, 2021          2021  TV-PG    41 min   
4  November 25, 2021          2021    NaN  1 Season   

                       listed_in  \
0              Animation, Family   
1                         Comedy   
2      Animation, Comedy, Family   
3                        Musical   
4  Docuseries, Historical, Music   

                                         description  
0   Join Mickey and the gang as they duck the halls!  
1  Santa Claus passes his magic bag to a new St. ...  
2          Sid the Sloth is on Santa's naughty list.  
3               This is real life, not just fantasy!  
4  A three-part documentary from Peter Jackson ca...

In [4]:
# Lets see the number of rows(sample) and columns(features)
df.shape

(1450, 12)

### In total we have 1450 Rows(Sample) and 12 Columns(Feature)

In [5]:
# Info function is used to get the information regarding the data and there datatypes
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1450 entries, 0 to 1449
Data columns (total 12 columns):
 #   Column        Non-Null Count  Dtype 
---  ------        --------------  ----- 
 0   show_id       1450 non-null   object
 1   type          1450 non-null   object
 2   title         1450 non-null   object
 3   director      977 non-null    object
 4   cast          1260 non-null   object
 5   country       1231 non-null   object
 6   date_added    1447 non-null   object
 7   release_year  1450 non-null   int64 
 8   rating        1447 non-null   object
 9   duration      1450 non-null   object
 10  listed_in     1450 non-null   object
 11  description   1450 non-null   object
dtypes: int64(1), object(11)
memory usage: 136.1+ KB


### We have 12 Features <br>
1. **show_id** - object datatype, id of the show or movie <br>
2. **type** - object datatype, Movie or TV Show <br>
3. **title** - object datatype, title of the movie <br>
4. **director** - Object Datatype, Director of the movie/tv show<br>
5. **cast** - Object Datatype, Cast of the movie/TV Show <br>
6. **country** - object datatype, Country where movie/tv show was released <br>
7. **date_added** - Object Datatype, Needs to be converted to datetime datatype, The date the movie was added to Disney + <br>
8. **release_year** - integer datatype, the year when the movie/tv show was released<br>
9. **rating** - object, Categorical data - The rating of the movie <br>
10. **duration** - object datatype, duration of movie in minutes and tv show in seasons<br>
11. **listed_in** - Object datatype, Categories the movie/tv show falls under. Eg: Animation, Comedy, Thriller, etc <br>
12. **description** - Object datatype, Description of movie/tv show <br>

In [6]:
# In here we have used isnull() to identify the total number of null value with the sum()
df.isnull().sum()

show_id           0
type              0
title             0
director        473
cast            190
country         219
date_added        3
release_year      0
rating            3
duration          0
listed_in         0
description       0
dtype: int64

### Signal to Noise Ratio(SNR)

In [7]:
# In here we have used the UNCLEAN Data to find the Signal To Noise Ratio(SNR)

# Finding the total signal power by squaring the total signal
total_signal = 1450
total_signal_power = np.mean(total_signal**2)

# Find the null values of individual columns
director_noise = df['director'].isnull().sum()
cast_noise = df['cast'].isnull().sum()
country_noise = df['country'].isnull().sum()
date_added_noise = df['date_added'].isnull().sum()
rating_noise = df['rating'].isnull().sum()

# Calculate the Noise of each by squaring the noise values from the above
director_noise_power = np.mean(director_noise**2)
cast_noise_power = np.mean(cast_noise**2)
country_noise_power = np.mean(country_noise**2)
date_added_noise_power = np.mean(date_added_noise**2)
rating_noise_power = np.mean(rating_noise**2)

# Finally we get the SNR by dividing the total signal with the individual feature(Column)
director_noise_snr = total_signal_power / director_noise_power
cast_noise_snr = total_signal_power / cast_noise_power
country_noise_snr = total_signal_power / country_noise_power
date_noise_snr = total_signal_power / date_added_noise_power
rating_noise_snr = total_signal_power / rating_noise_power

# We need to make the noise data more relevant so we have converted that to the Decibels
director_snr_db = 10 * np.log10(director_noise_snr)
cast_snr_db = 10 * np.log10(cast_noise_snr)
country_snr_db = 10 * np.log10(country_noise_snr)
date_snr_db = 10 * np.log10(date_noise_snr)
rating_snr_db = 10 * np.log10(rating_noise_snr)

# Printing Statements
print("director SNR:", director_noise_snr)
print("director SNR DB:", director_snr_db)

print("\ncast SNR:", cast_noise_power)
print("cast SNR DB:", cast_snr_db)

print("\ncountry SNR:", country_noise_power)
print("country SNR DB:", country_snr_db)

print("\ndate SNR:", date_added_noise_power)
print("date SNR DB:", date_snr_db)

print("\nrating SNR:", rating_noise_power)
print("rating SNR DB:", rating_snr_db)

director SNR: 9.39753004751284
director SNR DB: 9.730137229943267

cast SNR: 36100.0
cast SNR DB: 17.652288025642918

country SNR: 47961.0
country SNR DB: 16.418477747897132

date SNR: 9.0
date SNR DB: 53.68493495030626

rating SNR: 9.0
rating SNR DB: 53.68493495030626


In [8]:
# Lets see the date_added feature, it is of object datatype we need to change it to datetime
df['date_added']

0        November 26, 2021
1        November 26, 2021
2        November 26, 2021
3        November 26, 2021
4        November 25, 2021
               ...        
1445          June 4, 2021
1446         April 2, 2021
1447     December 18, 2020
1448    September 18, 2020
1449         April 1, 2020
Name: date_added, Length: 1450, dtype: object

### We will now work toward's the continuity of data by handling null values

In [9]:
# We need to use the strip module first because some values in this dataset still contain spaces at the beginning
# or end of string.
df["date_added"] = df["date_added"].str.strip()

# convert datatype to datetime
df["date_added"] = pd.to_datetime(df['date_added'])

# extract month and year
df['month_added']=df['date_added'].dt.month_name()
df['year_added'] = df['date_added'].dt.year

# Converting the year_added field to INT from FLOAT
df['year_added'] = df['year_added'].fillna(0).astype(int)

Here we can see the date_added is in the form of - **Month date, year** <br>
Here we can see the year_added is in the form of - **Float** <br>

We need to fix this <br>

Lets also make 2 more columns - **Month added and Year added **<br>
We can extract this from our **data_added** feature<br>

As you can see the **data_added** format has been changed <br>
There are also two new columns - **month_added, year_added**

In [10]:
# Showing the changed data type
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1450 entries, 0 to 1449
Data columns (total 14 columns):
 #   Column        Non-Null Count  Dtype         
---  ------        --------------  -----         
 0   show_id       1450 non-null   object        
 1   type          1450 non-null   object        
 2   title         1450 non-null   object        
 3   director      977 non-null    object        
 4   cast          1260 non-null   object        
 5   country       1231 non-null   object        
 6   date_added    1447 non-null   datetime64[ns]
 7   release_year  1450 non-null   int64         
 8   rating        1447 non-null   object        
 9   duration      1450 non-null   object        
 10  listed_in     1450 non-null   object        
 11  description   1450 non-null   object        
 12  month_added   1447 non-null   object        
 13  year_added    1450 non-null   int32         
dtypes: datetime64[ns](1), int32(1), int64(1), object(11)
memory usage: 153.1+ KB


**date_added** has been changed to datetime <br>
**month_added** - object datatype, Gives us information about the month the movie/tv show was added to Disney + <br>
**year_added** - int datatype, Gives us the information about the year the movie/tv show was added <br>

**We have some null values in our dataset** <br>
- **director**: There are **473** null values.
- **cast**: There are **190 null values**
- **country** - There are **219 null values**
- **date_added** - There are **3 null values**
- **rating** - There are **3 null values**
- **month_added** - There are **3 null values**
- **year_added** - There are **3 null values**

In [11]:
df.describe()

date_added  release_year  year_added
count                           1447   1450.000000  1450.00000
mean   2020-05-22 01:54:26.620594176   2003.091724  2015.52069
min              2019-10-01 00:00:00   1928.000000     0.00000
25%              2019-11-12 00:00:00   1999.000000  2019.00000
50%              2019-11-12 00:00:00   2011.000000  2019.00000
75%              2020-11-26 00:00:00   2018.000000  2020.00000
max              2021-11-26 00:00:00   2021.000000  2021.00000
std                              NaN     21.860162    91.80801

Describe shows us the statistical information of our numerical data<BR>
1. There are **1450** rows/samples in **release_year**<br>
- The earliest Movie released that is available to stream on Disney + is **1928** <br>
- The latest movie released that is available to stream on disney + was released in **2021**
2. There are **1447** records in **date_added**  <br>
- The **first** movie added to Disney + was on **2019-10-01**
- The **latest** movie added to Disney + was on **2021-11-26**
3. There are **1447** records in **year_added**

- **director**, **cast**, and **country** have too many null values so we can not drop it. It will have an adverse effect on our dataset. <br>
- It will be useful to have information about the director, and cast<br>
- We can replace the blank countries with the **mode** - the most common country <br>
- We can drop the 3 null records of the date_added; this will take care of the null values of month_added and year_added. As both those columns are derived from date_added

In [12]:
# Lets take care of the country feature, we will add the mode to all the missing values
df['country'] = df['country'].fillna(df['country'].mode()[0])

In [13]:
# keep cast as it could be interesting to look at a certain cast's films
df['cast'].replace(np.nan, 'No Data', inplace =True)
# keep director as it could be interesting to look at a certain director's films
df['director'].replace(np.nan, 'No Data', inplace = True)

In [14]:
# lets check the null values we have
df.isnull().sum()

show_id         0
type            0
title           0
director        0
cast            0
country         0
date_added      3
release_year    0
rating          3
duration        0
listed_in       0
description     0
month_added     3
year_added      0
dtype: int64

In [15]:
# Lets check for duplicated values to see if the date_added, rating and month_added does not have the duplicate values
df.duplicated().sum()

0

We have no duplicate values in our dataset <br>
We have taken care the null values in cast, country and director <br>
Now lets drop the remaining null records(sample)

In [16]:
# Droping the remaining nine rows with null values
df.dropna(inplace = True)

In [17]:
# Checking for the null value
df.isnull().sum()

show_id         0
type            0
title           0
director        0
cast            0
country         0
date_added      0
release_year    0
rating          0
duration        0
listed_in       0
description     0
month_added     0
year_added      0
dtype: int64

We now have 0 NULL Values

In [18]:
df.info()

<class 'pandas.core.frame.DataFrame'>
Index: 1444 entries, 0 to 1449
Data columns (total 14 columns):
 #   Column        Non-Null Count  Dtype         
---  ------        --------------  -----         
 0   show_id       1444 non-null   object        
 1   type          1444 non-null   object        
 2   title         1444 non-null   object        
 3   director      1444 non-null   object        
 4   cast          1444 non-null   object        
 5   country       1444 non-null   object        
 6   date_added    1444 non-null   datetime64[ns]
 7   release_year  1444 non-null   int64         
 8   rating        1444 non-null   object        
 9   duration      1444 non-null   object        
 10  listed_in     1444 non-null   object        
 11  description   1444 non-null   object        
 12  month_added   1444 non-null   object        
 13  year_added    1444 non-null   int32         
dtypes: datetime64[ns](1), int32(1), int64(1), object(11)
memory usage: 163.6+ KB


So we have cleaned our data <br>
- There are **1444** **Samples**(rows)
- There are **14** **Features**(columns)

We have 14 Features Now <br>
1. **show_id** - object datatype, id of the show or movie <br>
2. **type** - object datatype, Movie or TV Show <br>
3. **title** - object datatype, title of the movie <br>
4. **director** - Object Datatype, Director of the movie/tv show<br>
5. **cast** - Object Datatype, Cast of the movie/TV Show <br>
6. **country** - object datatype, Country where movie/tv show was released <br>
7. **date_added** - Datetime Datatyp, The date the movie was added to Disney + <br>
8. **release_year** - integer datatype, the year when the movie/tv show was released<br>
9. **rating** - object, Categorical data - The rating of the movie <br>
10. **duration** - object datatype, duration of movie in minutes and tv show in seasons<br>
11. **listed_in** - Object datatype, Categories the movie/tv show falls under. Eg: Animation, Comedy, Thriller, etc <br>
12. **description** - Object datatype, Description of movie/tv show <br>
13. **month_added** - object datatype, Gives us information about the month the movie/tv show was added to Disney + <br>
14. **year_added** - int datatype, Gives us the information about the year the movie/tv show was added <br>

### Explaining manifold number of dimensions of the underlying process that generated the dataset<br>
- In the Manifold dimension include number of process that resulted in dataset generation. The dataset you provided contains information about movies and TV shows, including attributes like title, director, cast, release date etc. <br>

- To identify the manifold dimension we have identify the unique feature and attribute of the each row in dataset. we can use the type, title, director, cast, and other feature as the potential dimesion.

- For example, If we have the the type we can categorically divide the data in how many of them are movies and how many of them are TV-Series. Another instance for this is, using the temporal value which is year and dates can be used to identify the latest trends pattern from this.


### Some of the question with the answers

In [19]:
#Q. Year in which maximun and minimun number of movies and series were released?

ans1 = df.groupby(['release_year']).size().idxmax()
print('Max movies and series released in :', ans1)
print("Maximum Count for the year is: ", df['release_year'].value_counts().max())

ans2 = df.groupby(['release_year']).size().idxmin()
print('Min movies and series released in :', ans2)
print("Maximum Count for the year is: ", df['release_year'].value_counts().min())


Max movies and series released in : 2021
Maximum Count for the year is:  123
Min movies and series released in : 1928
Maximum Count for the year is:  1


### Q. Year in which maximun and minimun number of movies and series were released?
Maximum number of movies and series were released in  the year : 2021 <br>
Maximum Count for the year 2021 is:  123 <br>
Minimum number of movies and series were released in the year : 1928 <br>
Maximum Count for the year 1928 is:  1 <br>

In [22]:
# Lets see the movie or tv show released in 1928
df[df['release_year'] == 1928]

show_id   type             title                director         cast  \
1220   s1221  Movie  Steamboat Willie  Walt Disney, Ub Iwerks  Walt Disney   

            country date_added  release_year rating duration  \
1220  United States 2019-11-12          1928  TV-Y7    8 min   

                    listed_in  \
1220  Animation, Family, Kids   

                                            description month_added  \
1220  Watch Mickey Mouse's classic cartoon debut. Se...    November   

      year_added  
1220        2019

In 1928, 1 movie was released - 'Steamboat Willie'. This movie was added in the year 2019

In [23]:
# Lets see the movies released in the year 2021
df[(df['type'] == 'Movie') & (df['release_year'] == 2021)]

show_id   type                                             title  \
3        s4  Movie                        The Queen Family Singalong   
5        s6  Movie                                 Becoming Cousteau   
14      s15  Movie                             America the Beautiful   
15      s16  Movie                                           Baymax!   
16      s17  Movie                                      Ciao Alberto   
..      ...    ...                                               ...   
266    s267  Movie  Star Wars Vintage: Story of the Faithful Wookiee   
282    s283  Movie                                      Own the Room   
290    s291  Movie                               Myth: A Frozen Tale   
297    s298  Movie                                   Flora & Ulysses   
301    s302  Movie                          Marvel's Behind the Mask   

                               director  \
3                       Hamish Hamilton   
5                            Liz Garbus   
14                              No Data   
15                              No Data   
16                       McKenna Harris   
..                                  ...   
266                             No Data   
282  Cristina Costantini, Darren Foster   
290                         Jeff Gipson   
297                           Lena Khan   
301                             No Data   

                                                  cast        country  \
3    Darren Criss, Adam Lambert, Derek Hough, Alexa...  United States   
5                Jacques Yves Cousteau, Vincent Cassel  United States   
14                                   Michael B. Jordan  United States   
15                                             No Data  United States   
16                 Jack Dylan Grazer, Marco Barricelli  United States   
..                                                 ...            ...   
266                                            No Data  United States   
282  Alondra Toledo, Daniela Blanco, Henry Onyango,...  United States   
290                                   Evan Rachel Wood  United States   
297  Matilda Lawler, Alyson Hannigan, Ben Schwartz,...  United States   
301                                            No Data  United States   

    date_added  release_year    rating duration                     listed_in  \
3   2021-11-26          2021     TV-PG   41 min                       Musical   
5   2021-11-24          2021     PG-13   94 min     Biographical, Documentary   
14  2021-11-12          2021     TV-PG    2 min              Animals & Nature   
15  2021-11-12          2021      TV-G    1 min                     Animation   
16  2021-11-12          2021     TV-PG    8 min     Animation, Comedy, Family   
..         ...           ...       ...      ...                           ...   
266 2021-04-02          2021  TV-Y7-FV    9 min                     Animation   
282 2021-03-12          2021     TV-14   91 min                   Documentary   
290 2021-02-26          2021      TV-G   12 min    Animation, Family, Fantasy   
297 2021-02-19          2021        PG   95 min  Comedy, Coming of Age, Drama   
301 2021-02-12          2021     TV-14   65 min        Anthology, Documentary   

                                           description month_added  year_added  
3                 This is real life, not just fantasy!    November        2021  
5    An inside look at the legendary life of advent...    November        2021  
14   Epic, grand portrait of the land across four r...    November        2021  
15         “Baymax!” premieres on Disney+ summer 2022.    November        2021  
16   Fun-loving sea monster Alberto whole-heartedly...    November        2021  
..                                                 ...         ...         ...  
266  With his friends ailing from a sleeping virus,...       April        2021  
282  Five young entrepreneurs from across the world...       March        2021  
290  A family sits down for a bedtime story

In [28]:
# Lets see the TV Shows released in the year 2021
df[(df['type'] == 'TV Show') & (df['release_year'] == 2021)]

show_id     type                                              title  \
6        s7  TV Show                                            Hawkeye   
18      s19  TV Show                                 Disney Intertwined   
28      s29  TV Show                                      Olaf Presents   
53      s54  TV Show                                       Storm Rising   
59      s60  TV Show                           The Wild Life of Dr. Ole   
63      s64  TV Show                                        Just Beyond   
66      s67  TV Show                                 The Wizard of Paws   
73      s74  TV Show                                    Among the Stars   
76      s77  TV Show  Drain The Oceans: The Mississippi River & Arct...   
78      s79  TV Show                              IMPACT with Gal Gadot   
79      s80  TV Show                          The Ghost and Molly McGee   
86      s87  TV Show                            Disney's Magic Bake-Off   
90      s91  TV Show                         Star Wars Galaxy of Sounds   
91      s92  TV Show                           The Hatcher Family Dairy   
97      s98  TV Show                     Spidey And His Amazing Friends   
98      s99  TV Show                                 Star Wars: Visions   
108    s109  TV Show                    Unknown Waters with Jeremy Wade   
112    s113  TV Show                             Doogie Kamealoha, M.D.   
119    s120  TV Show                                           Dug Days   
129    s130  TV Show                                  Growing Up Animal   
132    s133  TV Show     America's Funniest Home Videos: Animal Edition   
133    s134  TV Show                                      Chicken Squad   
134    s135  TV Show       Disney Presents Goofy in How to Stay at Home   
136    s137  TV Show                                        What If...?   
141    s142  TV Show              Cesar Millan: Better Human Better Dog   
145    s146  TV Show                           Chip 'n' Dale: Park Life   
148    s149  TV Show                              Mickey Mouse Funhouse   
149    s150  TV Show              Minnie's Bow-Toons: Party Palace Pals   
150    s151  TV Show              Turning the Tables with Robin Roberts   
158    s159  TV Show                              Behind the Attraction   
160    s161  TV Show       Meet Spidey and His Amazing Friends (Shorts)   
162    s163  TV Show                                     Turner & Hooch   
165    s166  TV Show                                 Shark Attack Files   
180    s181  TV Show                               Breaking Bobby Bones   
182    s183  TV Show                                   Monsters at Work   
189    s190  TV Show                    The Mysterious Benedict Society   
191    s192  TV Show                                         Dino Ranch   
201    s202  TV Show                                               Loki   
212    s213  TV Show                         Kingdom of the Polar Bears   
223    s224  TV Show                    Race to the Center of the Earth   
232    s233  TV Show                      Star Wars Vehicle Flythroughs   
233    s234  TV Show                                      The Bad Batch   
243    s244  TV Show                              Secrets of the Whales   
244    s245  TV Show                                           Big Shot   
245    s246  TV Show                                        Earth Moods   
265    s266  TV Show      Star Wars Vintage: Clone Wars 2D Micro-Series   
274    s275  TV Show                    The Mighty Ducks: Game Changers   
278    s279  TV Show                  The Falcon and The Winter Soldier   
293    s294  TV Show                         Secrets of Sulphur Springs   
310    s311  TV Show                                      Pixar Popcorn   
315    s316  TV Show                                        WandaVision   
318    s319  TV Show                             Marvel Studios LEGENDS   
332    s333  TV Show         

In [27]:
# Lets see the number of movies released in the year 2021
df[(df['type'] == 'TV Show') & (df['release_year'] == 2021)].shape

(53, 14)

### Q) What is the maximum and minimum number of movies and tv shows released?<br>

**In 2021**
- The most number of TV Shows and Movies released was in the year 2021
- 123 Movies/TV Shows were released in 2021
- In 2021, 70 Movies were released and 53 TV Shows were released <br>

**In 1928**
- The least number of movies and tv shows were released in the year |1928
- Only 1 Movie - "Streamboat Willie" was released in 1928

In [21]:
#Q. What was most popular genre of all time?
df['listed_in'].mode() 

0    Animation, Comedy, Family
Name: listed_in, dtype: object

### Q) What was most popular genre of all time?
The most Popular genre is Animation, Comedy and Family

In [29]:
# How many movies or series were released in each year?
df.groupby(['release_year']).size()

release_year
1928      1
1932      3
1933      3
1934      4
1935      4
       ... 
2017     69
2018     65
2019     98
2020    114
2021    123
Length: 90, dtype: int64

In [50]:
#Q how many movies and TV Shows were added each year to Disney Plus
df.groupby(['year_added']).size()

year_added
2019    765
2020    352
2021    327
dtype: int64

### Q) How many movies and TV Shows were added each year to Disney Plus? 
- In 2019 - 765 Movies/ TV Shows
- In 2020 - 352 Movies/ TV Shows
- In 2021 - 327 Movies/ TV Shows

In [30]:
# Q. What is the total number of the movie or TV show released?
df.groupby(['type']).size()

type
Movie      1051
TV Show     393
dtype: int64

### Q) What is the total number of the movie or TV show released?
**In Total** <br>
1051 Movies and 393 TV Shows 

In [31]:
ans1 = df.groupby(['year_added']).size().idxmax()
print('Max movies and series added in :', ans1)
print("Maximum Count for the year is: ", df['year_added'].value_counts().max())

ans2 = df.groupby(['year_added']).size().idxmin()
print('Min movies and series added in :', ans2)
print("Maximum Count for the year is: ", df['year_added'].value_counts().min())

Max movies and series added in : 2019
Maximum Count for the year is:  765
Min movies and series added in : 2021
Maximum Count for the year is:  327


### Q)Which year was the most and least number of Movies and TV Shows added?
In Disney Plus, the most number of Movies/TV Shows were added in 2019 and least number of Movies/TV Shows were added in 2021
- In 2019, 765 Movies/TV Shows were added
- In 2021, only 327 Movies/TV Shows were added<br>

In [38]:
#lets see the records of the Movies added in 2019
df[(df['year_added'] == 2019) & (df['type'] == 'Movie')].head()

show_id   type                                              title  \
674    s675  Movie              Forky Asks a Question: What is a Pet?   
675    s676  Movie            Star Wars: The Last Jedi (Episode VIII)   
676    s677  Movie           Forky Asks a Question: What is a Leader?   
677    s678  Movie                             George of the Jungle 2   
678    s679  Movie  High School Musical: The Musical: The Series: ...   

             director                                               cast  \
674      Bob Peterson              Tony Hale, Aloma Wright, Bob Peterson   
675      Rian Johnson  Mark Hamill, Carrie Fisher, Adam Driver, Daisy...   
676      Bob Peterson               Tony Hale, Bonnie Hunt, Bob Peterson   
677    David Grossman  Thomas Church, Julie Benz, Christina Pickles, ...   
678  Clayton Cogswell  Joshua Bassett, Olivia Rodrigo, Sofia Wylie, M...   

                      country date_added  release_year rating duration  \
674             United States 2019-12-27          2019   TV-G    7 min   
675             United States 2019-12-26          2017  PG-13  152 min   
676             United States 2019-12-20          2019   TV-G    6 min   
677  United States, Australia 2019-12-20          2003     PG   89 min   
678             United States 2019-12-20          2019   TV-G   23 min   

                             listed_in  \
674          Animation, Comedy, Family   
675  Action-Adventure, Science Fiction   
676          Animation, Comedy, Family   
677   Action-Adventure, Comedy, Family   
678  Coming of Age, Documentary, Drama   

                                           description month_added  year_added  
674  Forky meets Rib Tickles, and is schooled on th...    December        2019  
675  New heroes and galactic legends join to unlock...    December        2019  
676  Dolly teaches Forky about the qualities of a g...    December        2019  
677  George must rescue his family, then save the j...    December        2019  
678  An exclusive look into the making of High Scho...    December        2019

In [35]:
#lets see the number of records of the Movies added in 2019
df[(df['year_added'] == 2019) & (df['type'] == 'Movie')].shape

(630, 14)

In [88]:
#lets see the records of the TV Shows added in 2019
df[(df['year_added'] == 2019) & (df['type'] == 'TV Show')]

show_id     type                               title director  \
679     s680  TV Show                  Pick of the Litter  No Data   
684     s685  TV Show          One Day at Disney (Shorts)  No Data   
686     s687  TV Show               Disney The Lion Guard  No Data   
688     s689  TV Show  Disney Star vs. the Forces of Evil  No Data   
691     s692  TV Show                 Henry Hugglemonster  No Data   
...      ...      ...                                 ...      ...   
1406   s1407  TV Show                    Wild Yellowstone  No Data   
1412   s1413  TV Show            Wizards of Waverly Place  No Data   
1414   s1415  TV Show             Wolverine and The X-Men  No Data   
1417   s1418  TV Show                    X-Men: Evolution  No Data   
1434   s1435  TV Show                           Smart Guy  No Data   

                                                   cast  \
679                                             No Data   
684                                             No Data   
686   Max Charles, Diamond White, Dusan Brown, Attic...   
688                            Eden Sher, Adam McArthur   
691   Lara Miller, Tom Kenny, Chiara Zanni, Lori Ala...   
...                                                 ...   
1406                                     Bart Thomasson   
1412  Selena Gomez, David Henrie, Jake Austin, Jenni...   
1414  Steve Blum, Chris Edgerly, Jennifer Hale, Mich...   
1417  Noel Fisher, Vincent Gale, Christopher Judge, ...   
1434  Tahj Mowry, John Jones, Jason Weaver, Essence ...   

                                                country date_added  \
679                                       United States 2019-12-20   
684                                       United States 2019-12-06   
686                                       United States 2019-12-03   
688                                       United States 2019-12-01   
691   Ireland, United Kingdom, United States, South ... 2019-12-01   
...                                                 ...        ...   
1406                                            Germany 2019-11-12   
1412                                      United States 2019-11-12   
1414                                      United States 2019-11-12   
1417                                      United States 2019-11-12   
1434                                      United States 2019-10-01   

      release_year rating   duration                             listed_in  \
679           2019  TV-PG   1 Season  Animals & Nature, Docuseries, Family   
684           2019  TV-PG   1 Season         Anthology, Docuseries, Family   
686           2015   TV-Y  3 Seasons   Action-Adventure, Animation, Comedy   
688           2014  TV-Y7  4 Seasons   Action-Adventure, Animation, Comedy   
691           2012   TV-Y  2 Seasons              Animation, Fantasy, Kids   
...            ...    ...        ...                                   ...   
1406          2015  TV-PG   1 Season  Animals & Nature, Docuseries, Travel   
1412          2007   TV-G  4 Seasons        Comedy, Coming of Age, Fantasy   
1414          2008  TV-Y7   1 Season     Action-Adventure, Animation, Kids   
1417          2000  TV-Y7  4 Seasons     Action-Adventure, Animation, Kids   
1434          1996   TV-G  3 Seasons           Comedy, Coming of Age, Kids   

                                            description month_added  \
679   Watch an adorable group of six dogs on their q...    December   
684   Meet the diverse group of people behind some o...    December   
686   The Lion Guard has assembled and now they will...    December   
688    Star battles villains throughout the multiverse.    December   
691                   Hugglemonsters always find a way!    December   
...                                                 ...         ...   
1406  Wild Yellowstone captures the beauty and essen...    November   
1412  Alex, Justin, and Max Russo are not your ordin...    November   
1414  Wolverine and the X-Men must joi

In [37]:
#lets see the number of records of the TV Shows added in 2019
df[(df['year_added'] == 2019) & (df['type'] == 'TV Show')].shape

(135, 14)

In [90]:
#lets see the records of the Movies added in 2020
df[(df['year_added'] == 2020) & (df['type'] == 'Movie')].head()

show_id   type                      title            director  \
325    s326  Movie                     Burrow  Madeline Sharafian   
328    s329  Movie      Max Keeble's Big Move            Tim Hill   
329    s330  Movie                       Soul         Pete Docter   
330    s331  Movie  Arendelle Castle Yule Log             No Data   
331    s332  Movie   Buried Truth of the Maya             No Data   

                                                  cast        country  \
325                                            No Data  United States   
328  Alex D. Linz, Larry Miller, Jamie Kennedy, Zen...  United States   
329  Jamie Foxx, Tina Fey, Angela Bassett, Ahmir-Kh...  United States   
330                                            No Data  United States   
331                                     Josh Bernstein  United States   

    date_added  release_year rating duration                  listed_in  \
325 2020-12-25          2020      G    6 min          Animation, Family   
328 2020-12-25          2001     PG   88 min             Comedy, Family   
329 2020-12-25          2020     PG  107 min  Animation, Comedy, Family   
330 2020-12-18          2020   TV-G  180 min          Animation, Family   
331 2020-12-18          2020  TV-PG   48 min    Documentary, Historical   

                                           description month_added  year_added  
325  A rabbit attempts to dig a burrow and learns t...    December        2020  
328  After a week of creating mayhem, Max has to fa...    December        2020  
329  Jamie Foxx stars in Pixar’s "Soul," a comedy a...    December        2020  
330  Set the wintry mood for your holiday celebrati...    December        2020  
331  A high-tech archaeological team explores the h...    December        2020

In [91]:
#lets see the number of records of the Movies added in 2020
df[(df['year_added'] == 2020) & (df['type'] == 'Movie')].shape

(230, 14)

In [92]:
#lets see the number of records of the TV Shows added in 2020
df[(df['year_added'] == 2020) & (df['type'] == 'TV Show')].shape

(122, 14)

In [89]:
#lets see the records of the Movies added in 2021
df[(df['year_added'] == 2021) & (df['type'] == 'Movie')].head()

show_id   type                                             title  \
0      s1  Movie  Duck the Halls: A Mickey Mouse Christmas Special   
1      s2  Movie                            Ernest Saves Christmas   
2      s3  Movie                      Ice Age: A Mammoth Christmas   
3      s4  Movie                        The Queen Family Singalong   
5      s6  Movie                                 Becoming Cousteau   

                            director  \
0  Alonso Ramirez Ramos, Dave Wasson   
1                        John Cherry   
2                       Karen Disher   
3                    Hamish Hamilton   
5                         Liz Garbus   

                                                cast        country  \
0  Chris Diamantopoulos, Tony Anselmo, Tress MacN...  United States   
1           Jim Varney, Noelle Parker, Douglas Seale  United States   
2  Raymond Albert Romano, John Leguizamo, Denis L...  United States   
3  Darren Criss, Adam Lambert, Derek Hough, Alexa...  United States   
5              Jacques Yves Cousteau, Vincent Cassel  United States   

  date_added  release_year rating duration                  listed_in  \
0 2021-11-26          2016   TV-G   23 min          Animation, Family   
1 2021-11-26          1988     PG   91 min                     Comedy   
2 2021-11-26          2011   TV-G   23 min  Animation, Comedy, Family   
3 2021-11-26          2021  TV-PG   41 min                    Musical   
5 2021-11-24          2021  PG-13   94 min  Biographical, Documentary   

                                         description month_added  year_added  
0   Join Mickey and the gang as they duck the halls!    November        2021  
1  Santa Claus passes his magic bag to a new St. ...    November        2021  
2          Sid the Sloth is on Santa's naughty list.    November        2021  
3               This is real life, not just fantasy!    November        2021  
5  An inside look at the legendary life of advent...    November        2021

In [40]:
#lets see the number of records of the Movies added in 2021
df[(df['year_added'] == 2021) & (df['type'] == 'Movie')].shape

(191, 14)

In [41]:
#lets see the number of records of the TV Shows added in 2021
df[(df['year_added'] == 2021) & (df['type'] == 'TV Show')].shape

(136, 14)

### Q)How many Movies/TV Shows were added in the year 2019 and 2021?<br>

**In the year 2019**
- 630 Movies were added to Disney Plus
- 135 TV Shows were added to Disney Plus<br>

**In the year 2020**
- 230 Movies were added to Disney Plus
- 122 TV Shows were added to Disney Plus <br>

**In the year 2021**
- 191 Movies were added to Disney Plus
- 136 TV Shows were added to Disney Plus

### For the year 2021

In [53]:
# Lets see how many movies and TV Shows were added in the same year as they were released in 2021
df[(df['year_added'] == 2021) & (df['release_year'] == 2021)].head()

show_id     type                       title         director  \
3       s4    Movie  The Queen Family Singalong  Hamish Hamilton   
5       s6    Movie           Becoming Cousteau       Liz Garbus   
6       s7  TV Show                     Hawkeye          No Data   
14     s15    Movie       America the Beautiful          No Data   
15     s16    Movie                     Baymax!          No Data   

                                                 cast        country  \
3   Darren Criss, Adam Lambert, Derek Hough, Alexa...  United States   
5               Jacques Yves Cousteau, Vincent Cassel  United States   
6   Jeremy Renner, Hailee Steinfeld, Vera Farmiga,...  United States   
14                                  Michael B. Jordan  United States   
15                                            No Data  United States   

   date_added  release_year rating  duration                    listed_in  \
3  2021-11-26          2021  TV-PG    41 min                      Musical   
5  2021-11-24          2021  PG-13    94 min    Biographical, Documentary   
6  2021-11-24          2021  TV-14  1 Season  Action-Adventure, Superhero   
14 2021-11-12          2021  TV-PG     2 min             Animals & Nature   
15 2021-11-12          2021   TV-G     1 min                    Animation   

                                          description month_added  year_added  
3                This is real life, not just fantasy!    November        2021  
5   An inside look at the legendary life of advent...    November        2021  
6   Clint Barton/Hawkeye must team up with skilled...    November        2021  
14  Epic, grand portrait of the land across four r...    November        2021  
15        “Baymax!” premieres on Disney+ summer 2022.    November        2021

In [66]:
# Lets see the number of movies and TV Shows were added in the same year as they were released in 2021
df[(df['year_added'] == 2021) & (df['release_year'] == 2021)].shape

(122, 14)

In 2021, 122 Movies/TV Shows were added the same year they were released

In [70]:
# In the year 2021, lets find out how many movies were added in the same year they were released
df[(df['year_added'] == 2021) & (df['release_year'] == 2021) & (df['type'] == 'Movie')].shape

(70, 14)

In [46]:
# In the year 2021, lets find out how many TV Shows were added in the same year they were released
df[(df['year_added'] == 2021) & (df['release_year'] == 2021) & (df['type'] == 'TV Show')].shape

(52, 14)

In 2021, 70 Movies and 52 TV Shows were added the same year they were released

### For the year 2020

In [75]:
# Lets see the movies and TV Shows were added in the same year as they were released in 2021
df[(df['year_added'] == 2020) & (df['release_year'] == 2020)]

show_id     type                                        title  \
325    s326    Movie                                       Burrow   
326    s327  TV Show                      Cosmos: Possible Worlds   
327    s328  TV Show  Disney Gallery / Star Wars: The Mandalorian   
329    s330    Movie                                         Soul   
330    s331    Movie                    Arendelle Castle Yule Log   
..      ...      ...                                          ...   
630    s631    Movie                                    Lamp Life   
632    s633  TV Show                  Diary of a Future President   
641    s642    Movie      Forky Asks a Question: What is Reading?   
642    s643    Movie                                         Loop   
645    s646    Movie       Forky Asks a Question: What is Cheese?   

               director                                               cast  \
325  Madeline Sharafian                                            No Data   
326             No Data                                Neil deGrasse Tyson   
327             No Data  Jon Favreau, Dave Filoni, Pedro Pascal, Bryce ...   
329         Pete Docter  Jamie Foxx, Tina Fey, Angela Bassett, Ahmir-Kh...   
330             No Data                                            No Data   
..                  ...                                                ...   
630    Valerie LaPointe     Annie Potts, Jim Hanks, Ally Maki, Emily Davis   
632             No Data  Tess Romero, Selenis Leyva, Charlie Bushnell, ...   
641        Bob Peterson  Tony Hale, Addison Andrews, Mila Crespo, Imani...   
642        Erica Milsom  Madison Bandy, Christiano Delgado, Louis Gonza...   
645        Bob Peterson               Tony Hale, Jeff Garlin, Bob Peterson   

           country date_added  release_year rating   duration  \
325  United States 2020-12-25          2020      G      6 min   
326  United States 2020-12-25          2020  TV-14   1 Season   
327  United States 2020-12-25          2020  TV-PG  2 Seasons   
329  United States 2020-12-25          2020     PG    107 min   
330  United States 2020-12-18          2020   TV-G    180 min   
..             ...        ...           ...    ...        ...   
630  United States 2020-01-31          2020   TV-G     11 min   
632  United States 2020-01-17          2020  TV-PG  2 Seasons   
641  United States 2020-01-10          2020   TV-G      7 min   
642  United States 2020-01-10          2020     PG     12 min   
645  United States 2020-01-03          2020   TV-G      6 min   

                                listed_in  \
325                     Animation, Family   
326  Action-Adventure, Docuseries, Family   
327  Action-Adventure, Docuseries, Family   
329             Animation, Comedy, Family   
330                     Animation, Family   
..                                    ...   
630             Animation, Comedy, Family   
632         Comedy, Coming of Age, Family   
641             Animation, Comedy, Family   
642                     Animation, Family   
645             Animation, Comedy, Family   

                                           description month_added  year_added  
325  A rabbit attempts to dig a burrow and learns t...    December        2020  
326  COSMOS: POSSIBLE WORLDS continues Carl Sagan’s...    December        2020  
327  DISNEY GALLERY / STAR WARS: THE MANDALORIAN is...    December        2020  
329  Jamie Foxx stars in Pixar’s "Soul," a comedy a...    December        2020  
330  Set the wintry mood for your holiday celebrati...    December        2020  
..                                                 ...         ...         ...  
630  Lamp Life answers the question of what happene...     January        2020  
632  Elena Cañero-Reed will one day become the U.S....     January        2020  
641  The energetic Peas-n-a-Pod siblings teach Fork...     January        2020  
642  Adrift on a lake, two kids with different ways...     January        2020  
645  Buttercup, annoyed w

In [74]:
# Lets see the number of movies and TV Shows were added in the same year as they were released in 2021
df[(df['year_added'] == 2020) & (df['release_year'] == 2020)].shape

(83, 14)

In 2020, 83 Movies/TV Shows were added the same year they were released

In [76]:
# In the year 2020, lets find out how many movies were added in the same year they were released
df[(df['year_added'] == 2020) & (df['release_year'] == 2020) & (df['type'] == 'Movie')].shape

(56, 14)

In [77]:
# In the year 2020, lets find out how many TV Shows were added in the same year they were released
df[(df['year_added'] == 2020) & (df['release_year'] == 2020) & (df['type'] == 'TV Show')].shape

(27, 14)

In 2020, 56 Movies and 27 TV Shows were added the same year they were released

### For the year 2019

In [80]:
# Lets see how many movies and TV Shows were added in the same year as they were released in 2019
df[(df['year_added'] == 2019) & (df['release_year'] == 2019)].head()

show_id     type                                              title  \
674    s675    Movie              Forky Asks a Question: What is a Pet?   
676    s677    Movie           Forky Asks a Question: What is a Leader?   
678    s679    Movie  High School Musical: The Musical: The Series: ...   
679    s680  TV Show                                 Pick of the Litter   
680    s681    Movie                                               Togo   

             director                                               cast  \
674      Bob Peterson              Tony Hale, Aloma Wright, Bob Peterson   
676      Bob Peterson               Tony Hale, Bonnie Hunt, Bob Peterson   
678  Clayton Cogswell  Joshua Bassett, Olivia Rodrigo, Sofia Wylie, M...   
679           No Data                                            No Data   
680      Ericson Core  Willem Dafoe, Julianne Nicholson, Christopher ...   

           country date_added  release_year rating  duration  \
674  United States 2019-12-27          2019   TV-G     7 min   
676  United States 2019-12-20          2019   TV-G     6 min   
678  United States 2019-12-20          2019   TV-G    23 min   
679  United States 2019-12-20          2019  TV-PG  1 Season   
680  United States 2019-12-20          2019     PG   117 min   

                                      listed_in  \
674                   Animation, Comedy, Family   
676                   Animation, Comedy, Family   
678           Coming of Age, Documentary, Drama   
679        Animals & Nature, Docuseries, Family   
680  Action-Adventure, Animals & Nature, Family   

                                           description month_added  year_added  
674  Forky meets Rib Tickles, and is schooled on th...    December        2019  
676  Dolly teaches Forky about the qualities of a g...    December        2019  
678  An exclusive look into the making of High Scho...    December        2019  
679  Watch an adorable group of six dogs on their q...    December        2019  
680  An exhilarating and uplifting adventure set in...    December        2019

In [81]:
# Lets see how many movies and TV Shows were added in the same year as they were released in 2019
df[(df['year_added'] == 2019) & (df['release_year'] == 2019)].shape

(45, 14)

In 2019, 45 Movies/TV Shows were added the same year they were released

In [83]:
# In the year 2019, lets find out how many Movie were added in the same year they were released
df[(df['year_added'] == 2019) & (df['release_year'] == 2019) & (df['type'] == 'Movie')].shape

(32, 14)

In [84]:
# In the year 2019, lets find out how many TV Shows were added in the same year they were released
df[(df['year_added'] == 2019) & (df['release_year'] == 2019) & (df['type'] == 'TV Show')].shape

(13, 14)

In 2019, 32 Movies and 13 TV Shows were added the same year they were released

### Q)How many Movies/TV Shows were added to disney plus the same year they were released?
<br>

**In the year 2019**<br>
- 32 Movies and 13 TV Shows were added to disney plus the same year they were released

**In the year 2020**<br>
- 56 Movies and 27 TV Shows were added to disney plus the same year they were released
    
**In the year 2021** <br>
- 70 Movies were added to disney plus the same year they were released and 52 TV Shows were added to disney plus the same year they were released<br>


### This In Class Assignment 1 has been completed by:
**TEAM 5**<BR>
**Daivik Pelathur - C0938273 <br>
Dev Bhagat - c0937406 <br>
Amritpal Singh - c0937405**